### ISSUE TITLE
FLASH5 hasn't inherited sourceTerms\Deleptonize yet?

### ISSUE THREAD

**Issue by ranchu1** [ECP-Astro/FLASH5#603]
I'm planning to run a simulation with Deleptonize. 
When will sourceTerms\Deleptonize be inherited by FLASH5 from FLASH4?

**Comment by kweide** [comment]
The `Deleptonize` unit weas removed in this commit:
```
commit abc210992963196361c0d1213baf7654e66ed09e
Author: adubey64 <adubey64@gmail.com>
Date:   Mon Mar 25 15:28:45 2019 +0100

    removing unsupported physics
```

**Comment by jaharris87** [comment]
@kweide What does unsupported mean in this context? Is there a reason we can't bring this back?

**Comment by kweide** [comment]
Most of the implementation was removed earlier by @adubey64 
```
commit aded69be1743b89ca05e83984fbce86ff03a2a7b
Author: dubey <adubey@anl.gov>
Date:   Wed Oct 17 15:06:58 2018 -0500

    removed implementations for all source terms except burn, which already has iterators
```

**Comment by jaharris87** [comment]
Right, but this was this because they will never be part of FLASH5 or because we didn't want Units that had not been translated to the new iterators lying around?

**Comment by kweide** [comment]
I supposed the code should just be (re-)added. Or, to be more specific, add the most recent or up to date version that exists for FLASH4.

**Comment by jaharris87** [comment]
Ok, this is the path forward we were thinking. @ranchu1 was going to give this a go, but we wanted to document it as an issue first.

**Comment by kweide** [comment]
@jaharris87 I thinks it was the latter, maybe @adubey64 can confirm.

**Comment by adubey64** [comment]
We didn't want units that won't work with new iterators lying around.

**Comment by jaharris87** [comment]
Thank you @kweide and @adubey64 for the quick responses.

As I see it, these are the things that need the happen to get Deleptonize in FLASH5:

- [ ] Restore `Deleptonize` to `sourceTerms` and update to use new iterators
- [ ] Restore `Driver_sourceTerms` to `Driver_evolveFlash` and remove call to `Burn`
- [ ] Add `Burn` and `Deleptonize` to `Driver_sourceTerms`
- [ ] Implement a regression test that uses `Deleptonize` unit.
